linode/manager
# feat: [M3-8883] - Use our search parser on Images Landing

## Description 📝

- Uses our search parser on the Images Landing page to provide users with slightly more flexibility  🔍
- This is just a "soft-launch".  We can add user facing documentation as a follow up once we're confident in the feature and the API has had some time to improve filterability.

## Preview 📷

### Search by `region`

This will help power users using Image Service Gen2

> [!note]
> Sadly does not support complex operations (`+or` / `+and`) due to API limitations. Only available in dev right now.

![screenshot](https://github.com/user-attachments/assets/4ba913eb-b5a7-4764-83fa-5b7dd9018818)

### Search by `label` and/or `size`
> [!note]
> Because the API returns size in MB, the user would have to search in MBs 🫤 Planning to bring this to UX, but not blocking in my opinion

![Screenshot 2024-11-08 at 1 02 38 PM](https://github.com/user-attachments/assets/cb5558d0-e6cb-4e58-b543-a361ea4ae739)

### Search by `tags` (or `tag` because it is aliased)

Adds a sustainable way to filter by tag 🎉 

> [!note]
> Searching by tag is partially broken. See `ARB-5792`

![Screenshot 2024-11-08 at 1 05 38 PM](https://github.com/user-attachments/assets/2210d34b-a322-482b-87a0-e42f6ef61996)


## How to test 🧪

- Test various search queries on the Images landing page 🔍 
- Verify the UI gracefully handles valid and invalid searches ✅ ❌ 

## As an Author I have considered 🤔
